assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -56,8 +56,17 @@ function MediaProvider({ children }) {
   const {
     actions: { getMedia },
   } = useAPI();
+
+  // Used by callbacks to avoid unnecessary dependency on state changes,
+  // which cause re-renderings.
+  const stateRef = useRef();
+  stateRef.current = state;
+
   const fetchMedia = useCallback(
     ({ pagingNum: p = 1, mediaType: currentMediaType } = {}, callback) => {
+      // eslint-disable-next-line no-shadow
+      const { searchTerm } = stateRef.current;
```

Comment:
I'd ask an earlier question here: why do we need this method at all? This looks like a completely internal thing to this media provider and shouldn't be exposed in this form to outside world.

Reply:
I think it's not completely internal, it's also used by CanvasUploadDropTarget (which uses uploadWithPreview that uses uploadMedia).

Reply:
I think that dependency is completely wrong. What it looks like it's trying to do is simply trigger reload. We should just have that API exported and it'd likely just be a `useState(state + 1)` here to trigger reload?

Reply:
'searchTerm' can be deleted from here after https://github.com/google/web-stories-wp/pull/1729 has been submitted.

Reply:
I've rebased this PR on top of #1729 and #2071, PTAL.